### ISSUE TITLE
missing_docs does not handle enum definitions

### ISSUE THREAD

**Issue by chadmoone** [realm/SwiftLint#658]
I don't see any examples in the `triggeringExamples` or `nonTriggeringExamples` pertaining to enums, and as far as I can tell, there is no way to document an enum that will prevent a warning.

Triggers warning (note, this seems to be the style Apple prefers/uses):

``` swift
/// The types of foo.
public enum Foo {

    /// The bar type.
    case Bar

    /// The baz type.
    case Baz
}
```

Also triggers:

``` swift
/**
 The types of foo.

 - Bar: The bar type.
 - Baz: The baz type.
 */
public enum Foo {
    case Bar
    case Baz
}
```

**Comment by jpsim** [comment]
I can't reproduce the missing docs violation you're seeing with your first example. What version of SwiftLint are you using?

**Comment by CBeloch** [comment]
I have the same issue with 0.10.0

**Comment by jpsim** [comment]
@CBeloch care to share a repro case? I can't reproduce this:

```
$ swiftlint version
0.10.0
$ cat .swiftlint.yml 
opt_in_rules:
  - missing_docs
$ cat foo.swift 
/// The types of foo.
public enum Foo {

    /// The bar type.
    case Bar

    /// The baz type.
    case Baz
}
$ swiftlint lint --path foo.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at path foo.swift
Linting 'foo.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
$ cat foo2.swift 
/**
 The types of foo.

 - Bar: The bar type.
 - Baz: The baz type.
 */
public enum Foo {
    case Bar
    case Baz
}
$ swiftlint lint --path foo2.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at path foo2.swift
Linting 'foo2.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

**Comment by CBeloch** [comment]
The problem is, that I activated the docs warnings for everything with an _internal_ ACL.
If you change your sample to
```swift
public enum Foo {
    /// The bar type.
    public case Bar
    /// The baz type.
    public case Baz
}
```
Or add this to your .swiftlint.yml:
```yaml
missing_docs:
  - internal
```
You will get the same error for all cases.

Why I want the missing_docs rule to work on internal types? Because I generate a code documentation with jazzy for the App itself.

**Comment by jpsim** [comment]
Still can't reproduce this @CBeloch:

```
$ swiftlint version
0.11.0
$ cat .swiftlint.yml
opt_in_rules:
  - missing_docs
missing_docs:
  - internal
$ cat foo.swift
public enum Foo {
    /// The bar type.
    public case Bar
    /// The baz type.
    public case Baz
}
$ swiftlint
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'foo.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

**Comment by CBeloch** [comment]
Well you haven't noted the OR in my last comment, did you?

```bash
$ swiftlint version
0.11.0

$ cat .swiftlint.yml
opt_in_rules:
  - missing_docs
missing_docs:
  - internal

$ cat foo.swift
enum Foo {
    /// The bar type.
    case Bar
    /// The baz type.
    case Baz
}

$ swiftlint
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'foo.swift' (1/1)
foo.swift:3:10: warning: Missing Docs Violation: Public declarations should be documented. (missing_docs)
foo.swift:5:10: warning: Missing Docs Violation: Public declarations should be documented. (missing_docs)
foo.swift:1:1: warning: Missing Docs Violation: Public declarations should be documented. (missing_docs)
Done linting! Found 3 violations, 0 serious in 1 file.

**Comment by CBeloch** [comment]
No progress here?

**Comment by jpsim** [comment]
I don't recall how this ended, but apparently I failed to update this thread. I'll reopen in the hopes to look into it again in the future.

**Comment by jpsim** [comment]
Looks like a bug to me.

**Comment by fergalrooney** [comment]
I have tons of enums in my applications leading to multiple incorrect warnings. Would be great to get this fixed!

**Comment by jeeftor** [comment]
Same issue for me.

**Comment by marcelofabri** [comment]
Closing this since we removed this rule.
